src/pipeline.rs

```diff
@@ -127,15 +151,16 @@ mod test {
             false,
         )?;
 
-        let (pipeline, reciever) =
+        let (mut pipeline, receiver) =
             Pipeline::new_with_buffer(100, vec![Box::new(transform_1), Box::new(transform_2)]);
 
         let event = Event::try_from(json!({
             "message": "MESSAGE_MARKER",
         }))?;
 
-        pipeline.send(event).compat().await?;
-        let out = reciever.wait().collect::<Result<Vec<_>, ()>>().unwrap();
+        pipeline.send(event).await?;
+        pipeline.flush().await?;
```

Comment:
I do not think that we really need `flush` because [send](https://docs.rs/futures/0.3.8/futures/sink/trait.SinkExt.html#method.send) already include flushing:
> A future that completes after the given item has been fully processed into the sink, **including flushing**.